string message
float32 modulo_lin
float32 modulo_ang